providers/connectivity.rb

```diff
@@ -89,64 +95,25 @@
    Chef::Log.debug("connectivity.rb ::: https_proxy_host => #{https_proxy_host}")
    Chef::Log.debug("connectivity.rb ::: https_proxy_port => #{https_proxy_port}")
 
-   if !(http_proxy_host.empty? && https_proxy_host.empty?)
-       # Pinging Proxy
-       proxy_ping = "ping -q -w 1 -c 1 #{http_proxy_host} > /dev/null"
-       ssl_proxy_ping = "ping -q -w 1 -c 1 #{https_proxy_host} > /dev/null"
-
-       proxy_cmd = shell_out("#{proxy_ping}").exitstatus == 0
-       ssl_proxy_cmd = shell_out("#{ssl_proxy_ping}").exitstatus == 0
-       Chef::Log.debug("connectivity.rb ::: proxy_cmd => #{proxy_cmd}") 
-       Chef::Log.debug("connectivity.rb ::: ssl_proxy_cmd => #{ssl_proxy_cmd}") 
-  
-       test_ok = (proxy_cmd or ssl_proxy_cmd)
-       Chef::Log.debug("connectivity.rb ::: proxy test_ok => #{test_ok}") 
+   # Gecos Control Center (GCC)
+   wget_opts = case scheme 
+       when 'http';  "-q"
+       when 'https'; "-q --no-check-certificate"
    end
-
-   # Pinging Gecos Control Center (GCC)
-   if (test_ok)
-       ping = "ping -q -w 1 -c 1 #{target} > /dev/null"
-       wget = "wget -q --no-check-certificate #{target}:#{port} -O /dev/null"
-   
-       ping_cmd = shell_out("#{ping}").exitstatus == 0
-       wget_cmd = shell_out("#{wget}", :env => {'http_proxy' => ENV['HTTP_PROXY'],'https_proxy' => ENV['HTTPS_PROXY']}).exitstatus == 0
-
-       test_ok = (ping_cmd && wget_cmd)
-	
-       node.normal['gcc_link'] = if test_ok
-           true
-       else
-          false
-       end
-       Chef::Log.debug("connectivity.rb ::: GCC test_ok => #{test_ok}") 
+   wget_cmd = "wget #{wget_opts} #{scheme}://#{host}:#{port} -O /dev/null"
```

Comment:
You could use some python library to perform HTTP requests, such as urllib, to improve error handling or page parsing, but in this case, only connectivity is checked. At the level of performance, and being the download of a few kilobytes, there is not much difference